ui/shared/AddressInput.qml

```diff
@@ -11,47 +11,63 @@ Item {
     //% "ENS Username not found"
     property string ensAsyncValidationError: qsTrId("ens-username-not-found")
     property alias label: inpAddress.label
+    property alias text: inpAddress.text
     property string selectedAddress
     property var isValid: false
     property bool isPending: false
+    readonly property string uuid: Utils.uuid()
+    property alias readOnly: inpAddress.readOnly
+    property bool isResolvedAddress: false
 
     height: inpAddress.height
 
+    onSelectedAddressChanged: validate()
+    onTextChanged: resolveEns()
+
     function resetInternal() {
         selectedAddress = ""
         inpAddress.resetInternal()
         metrics.text = ""
         isValid = false
+        isPending = false
+        isResolvedAddress = false
+    }
+
+    function resolveEns() {
+        if (Utils.isValidEns(text)) {
+            root.validateAsync(text)
+        }
     }
 
-    function validate(inputValue) {
-        if (!inputValue) inputValue = selectedAddress
-        let isValid =
-            (inputValue && inputValue.startsWith("0x") && Utils.isValidAddress(inputValue) || Utils.isValidEns(inputValue))
-        inpAddress.validationError = isValid ? "" : validationError
+    function validate() {
+        let isValidEns = Utils.isValidEns(text)
+        let isValidAddress = Utils.isValidAddress(selectedAddress)
+        let isValid = (isValidEns && !isResolvedAddress) || isPending || isValidAddress
+        inpAddress.validationError = ""
+        if (!isValid){ 
+            inpAddress.validationError = isResolvedAddress ? ensAsyncValidationError : validationError
+        }
         root.isValid = isValid
         return isValid
     }
 
-    property var validateAsync: Backpressure.debounce(inpAddress, 300, function (inputValue) {
+    property var validateAsync: Backpressure.debounce(inpAddress, 600, function (inputValue) {
         root.isPending = true
+        root.selectedAddress = ""
         var name = inputValue.startsWith("@") ? inputValue.substring(1) : inputValue
-        walletModel.resolveENS(name)
+        walletModel.resolveENS(name, uuid)
     });
 
 
     Connections {
         target: walletModel
         onEnsWasResolved: {
-            root.isPending = false
-            if (resolvedPubKey === ""){
-                inpAddress.validationError = root.ensAsyncValidationError
-                root.isValid = false
-            } else {
-                root.isValid = true
-                root.selectedAddress = resolvedPubKey
-                inpAddress.validationError = ""
+            if (uuid !== root.uuid) {
```

Comment:
I understand why you used a `uuid`, but just using the ens name as a differentiator would have worked as well no?
```
if (name !== root.name) {
```

Reply:
Unfortunately not. Because we use a `Connections` component that listens for the resolution response in multiple *other* components, each and every connection would react to resolutions that it did not request.